jobs/upload-brew.Jenkinsfile

```diff
@@ -0,0 +1,128 @@
+    ]),
+    buildDiscarder(logRotator(
+        numToKeepStr: '100',
+        artifactNumToKeepStr: '100'
+    )),
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+])
+
+def build_description = "[${params.VERSION}]"
+
+// Always use the latest
+cosa_img = "quay.io/coreos-assembler/coreos-assembler:latest"
+
+// Note the supermin VM just uses 2G. The really hungry part is xz, which
+// without lots of memory takes lots of time. For now we just hardcode these
+// here; we can look into making them configurable through the template if
+// developers really need to tweak them.
+// XXX bump an extra 2G (to 10.5) because of an error we are seeing in
+// testiso: https://github.com/coreos/fedora-coreos-tracker/issues/1339
+def cosa_memory_request_mb = 10.5 * 1024 as Integer
+
+// Now that we've established the memory constraint based on xz above, derive
+// kola parallelism from that. We leave 512M for overhead and VMs are at most
+// 1.5G each (in general; root reprovisioning tests require 4G which is partly
+// why we run them separately).
+// XXX: https://github.com/coreos/coreos-assembler/issues/3118 will make this
+// cleaner
+def ncpus = ((cosa_memory_request_mb - 512) / 1536) as Integer
+
+echo "Waiting for build-upload-to-brew lock"
+currentBuild.description = "${build_description} Waiting"
+def stream_info = pipecfg.streams[params.STREAM]
+
+lock(resource: "build-upload-to-brew") {
+    timeout(time: 240, unit: 'MINUTES') {
+    cosaPod(cpu: "${ncpus}",
+            memory: "${cosa_memory_request_mb}Mi",
+            image: cosa_img,
+            serviceAccount: "jenkins") {
+    try {
+        stage('Fetch Metadata') {
+            def ref = pipeutils.get_source_config_ref_for_stream(pipecfg, params.STREAM)
+            def variant = stream_info.variant ? "--variant ${stream_info.variant}" : ""
+            def s3_stream_dir = pipeutils.get_s3_streams_dir(pipecfg, params.STREAM)
+            pipeutils.shwrapWithAWSBuildUploadCredentials("""
+                cosa init --branch ${ref} ${variant} ${pipecfg.source_config.url}
+                cosa buildfetch --build=${params.VERSION} \
+                    --arch=${params.ARCH} --url=s3://${s3_stream_dir}/builds \
+                    --aws-config-file \${AWS_BUILD_UPLOAD_CONFIG} \
+                    --artifact=qemu \
+                    --file "coreos-assembler-config-git.json" 
+            """)
+        }
+        stage('Brew Upload') {
+            withCredentials([
+                file(credentialsId: 'brew-cfg', variable: 'BREW'),
+                     file(credentialsId: 'brew-krb-config', variable: 'KRB5'),
+                     file(credentialsId: 'brew-ca', variable: 'CA'),
+                     file(credentialsId: 'brew-keytab', variable: 'KEYTAB'),
+             ]) {
+                shwrap("""
+                    cp $BREW /etc/koji.conf
+                    cp $KRB5 /etc/krb5.conf
+                    cp $KEYTAB /tmp/keytab
+                    cp $CA /etc/pki/brew/brew-ca
```

Comment:
I assume there's no way to override the place to look for these files when invoking `brew`? That might be easier than modifying `cosa`.